### PR TITLE
added additional IR tests

### PR DESCRIPTION
**Pull request by cseed**
uncovered a few edge cases

### CODE REVIEW

File: src/test/scala/is/hail/expr/ir/IRSuite.scala [hunk 1]
```diff
@@ -103,6 +107,108 @@ class IRSuite extends TestNGSuite {
...
+
+    val a = MakeArray(FastIndexedSeq(I32(-7), NA(TInt32()), I32(2)), TArray(TInt32()))
+    assertEvalsTo(ArraySort(a),
+      FastIndexedSeq(-7, 2, null))
+  }
+
+  @Test def testToSet() {
+    assertEvalsTo(ToSet(NA(TArray(TInt32()))), null)
+
+    val a = MakeArray(FastIndexedSeq(I32(-7), NA(TInt32()), I32(2)), TArray(TInt32()))
```
Comment (jigold) [inline src/test/scala/is/hail/expr/ir/IRSuite.scala]: Can you add duplicate elements?
